transpiler/include/nil/blueprint/transpiler/aggregated_verifier_generator.hpp

```diff
@@ -0,0 +1,296 @@
+                            if (*rit != 0) { [[unlikely]]
+                                max_non_zero = std::distance(rit, public_inputs[i].rend()) - 1;
+                                break;
+                            }
+                        }
+                        if (max_non_zero + 1 > public_input_sizes[i]) {
+                            std::cout << "Public input size is larger than reserved. Real size = "
+                                      << max_non_zero  + 1 << " reserved = " << public_input_sizes[i] << std::endl;
+                            exit(1);
+                        }
+                        BOOST_ASSERT(max_non_zero <= public_input_sizes[i]);
+                        std::size_t j = 0;
+                        for (; j < public_inputs[i].size(); j++, after_first = true) {
+                            if (after_first) [[likely]] out << "," << std::endl;
+                            out << "\t\t{\"field\": \"" << public_inputs[i][j] << "\"}";
+                        }
+                        for (; j < public_input_sizes[i]; j++, after_first = true) {
+                            if (after_first) [[likely]] out << "," << std::endl;
+                            out << "\t\t{\"field\": \"" << typename field_type::value_type(0) << "\"}";
+                        }
+                    }
+                    out << std::endl << "\t]}," << std::endl;
+                }
+                // commitments for each partial proof
+                // aka std::vector<placeholder_partial_proof<FieldType, ParamsType>> partial_proofs;
+                for (auto const &partial_proof : proof.partial_proofs) {
+                    out << "\t{\"struct\":[" << std::endl;
+                    out << "\t\t{\"array\":[" << std::endl;
+                    bool after_first = false;
+                    for (const auto &[index, commitment]: partial_proof.commitments) {
+                        if (after_first) [[likely]] out << "," << std::endl;
+                        out << "\t\t\t"
+                            << generate_commitment<typename PlaceholderParams::commitment_scheme_type>(commitment);
+                        after_first = true;
+                    }
+                    out << "\t\t]}]}," << std::endl;
+                }
+                // aggregated proof type
+                const auto &aggregated_proof = proof.aggregated_proof;
+                out << "\t{\"struct\":[" << std::endl;
+                // single fri proof checking that F(x) is low degree
+
+                // basic_fri::round_proofs_batch_type fri_round_proof
+                const auto &fri_round_proof = aggregated_proof.fri_proof;
+                // which is in essence std::vector<std::vector<round_proof_type>> round_proofs;
+                out << "\t\t{\"array\":[" << std::endl;
+                bool after_first = false;
+                for (const auto &outer_proof_vector : fri_round_proof.fri_round_proof.round_proofs) {
```

Comment:
We also need round proof merkle paths. Only hashes. Not positions

Reply:
Should be fixed now.